src/components/CustomHeader/CustomHeader.tsx

```diff
@@ -0,0 +1,73 @@
+import { useEffect, useState } from 'react';
+import CustomButton from '../CustomButton/CustomButton';
+import { useNavigate } from 'react-router-dom';
+import { searchSlice } from '../../stores/reducers/SearchSlice';
+import { useAppDispatch, useAppSelector } from '../../stores/hooks/redux';
+import CustomInput from '../CustomInput/CustomInput';
+import '../components.css';
+
+export interface HeaderState {
+  handleStartSearch: () => void;
+  handleStopSearch: () => void;
+}
+
+const CustomHeader: React.FC<HeaderState> = (props) => {
+  const { storedSearchValue } = useAppSelector((state) => state.searchReducer);
+  const [inputValue, setInputValue] = useState(storedSearchValue || '');
```

Comment:
In your implementation the ```storedSearchValue``` already is a existing value or empty string, so you don't need to repeat it here: ```const [inputValue, setInputValue] = useState(storedSearchValue);```

Reply:
fixed